### PR TITLE
feat: migrate table config conditional formatting to mantine

### PR DESCRIPTION
**Pull request by magnew**
Closes: #6256 

### Description:

Moves table formatting config to Mantine.

New look:

Screenshot 2023-07-17 at 19 31 20

Old:

old

### Criteria
acceptance criteria:
- [ ] use pie charts series config styles as a reference (they kinda look the same)
- [ ] migrate rule container with collapsible element
- [ ] migrate condition container with collapsible element
- [ ] migrate all the buttons
- [ ] use new mantine color picker (reuse color picker from pie charts and extract it somewhere else)
- [ ] ignore operator selector and value input (that one is reused from "filters" which we'll tackle as a separate milestone)

### CODE REVIEW

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/index.tsx [hunk 1]
```diff
@@ -14,13 +16,23 @@ const TableConfigPanel: React.FC = () => {
...
+                >
+                    Configure
+                </Button>
+            </Popover.Target>
+
+            <Popover.Dropdown>
                 <Box
                     w={320}
-                    p="sm"
+                    p={0}
```
Comment (joaoviana) [inline packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/index.tsx:35]: I don't think we need this as well 🤔

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/index.tsx [hunk 2]
```diff
@@ -14,13 +16,23 @@ const TableConfigPanel: React.FC = () => {
...
                     sx={{
                         // FIXME: remove after Blueprint migration is complete
```
Comment (joaoviana) [inline packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/index.tsx:37]: could you check if this is necessary now? 🤔

**Review by joaoviana — CHANGES_REQUESTED** [review]
Looks great! Just some small observations on styles etc 👍

File: packages/frontend/src/components/VisualizationConfigs/ColorSelector.tsx [hunk 3]
```diff
@@ -73,4 +77,4 @@ const ColorPicker: FC<Props> = ({
     );
 };
 
-export default ColorPicker;
+export default ColorSelector;
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/VisualizationConfigs/ColorSelector.tsx:80]: Yup, good call, I forgot.

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/GeneralSettings.tsx [hunk 4]
```diff
@@ -168,20 +167,22 @@ const GeneralSettings: FC = () => {
...
                 position="top"
             >
-                <Checkbox
-                    disabled={!canUsePivotTable}
-                    label="Show metrics as rows"
-                    checked={metricsAsRows}
-                    onChange={() => handleToggleMetricsAsRows()}
-                />
-            </Tooltip2>
+                <Box>
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/GeneralSettings.tsx:177]: Yea. Or at least this is the way I know to make this work in Mantine. If there is another way I'm open to it. 

The problem is that if you disable the input, the tooltip gets disabled. In this case we want to show the tooltip *when* the checkbox is disabled. So you can put the tooltip on the box. I don't know if we want a comment here or we just need to know this. Comments in JSX are a bit annoying :). But I can add one.

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ConditionalFormatting.tsx [hunk 5]
```diff
@@ -210,24 +214,27 @@ const ConditionalFormatting: FC<ConditionalFormattingProps> = ({
                                 />
 
                                 {ruleIndex !== config.rules.length - 1 && (
-                                    <ConditionalFormattingRuleAndLabel>
-                                        AND
-                                    </ConditionalFormattingRuleAndLabel>
+                                    <Box p={0}>
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ConditionalFormatting.tsx:217]: Good call, I will remove. I think I had something more complex here before.

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ConditionalFormatting.tsx [hunk 6]
```diff
@@ -121,68 +129,64 @@ const ConditionalFormatting: FC<ConditionalFormattingProps> = ({
...
+                                    value: id,
+                                    label: fieldLabelText(f),
+                                    disabled:
+                                        id === (field && getItemId(field)),
+                                };
+                            })}
+                            itemComponent={FieldSelectItem}
+                            onChange={handleChangeField}
+                        />
+                        <Group spacing="xs" my="xs">
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ConditionalFormatting.tsx:182]: It looks a little crowded to me, but I don't have a strong opinion. Will do.

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/GeneralSettings.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/GeneralSettings.tsx:177]: Got it! ignore, no need for comment 👍

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/index.tsx (hunk 1 above)
Comment (magnew) [reply packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/index.tsx:35]: Done.

File: packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/index.tsx (hunk 2 above)
Comment (magnew) [reply packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/index.tsx:37]: The next review removes this too :)

**Review by joaoviana — APPROVED** [review]

**Comment by IrakliJani** [comment]
there's a render deployment bug, checking what's going on there

**Pull request merged**